pkg/workload/histogram/exporter/exporter_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package exporter
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	mockHistogramName = "mock_histogram"
+)
+
+func TestHdrJsonExporter(t *testing.T) {
+	exporter := &HdrJsonExporter{}
+	t.Run("Validate", func(t *testing.T) {
+		err := exporter.Validate("metrics.json")
+		assert.NoError(t, err)
+
+		err = exporter.Validate("metrics.txt")
+		assert.Error(t, err)
+		assert.Equal(t, "file path must end with .json", err.Error())
+	})
+
+	t.Run("Init", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.json")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+	})
+
+	t.Run("SnapshotAndWrite", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.json")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+
+		start := time.Time{}
+		elapsed := time.Second / 2
+		name := mockHistogramName
+		mockHistogram := hdrhistogram.New(0, 100, 1)
+		err = exporter.SnapshotAndWrite(mockHistogram, start, elapsed, &name)
+		require.NoError(t, err)
+		err = exporter.Close(nil)
+		require.NoError(t, err)
+
+		var data map[string]interface{}
+		buf, err := os.ReadFile(file.Name())
+		require.NoError(t, err)
+		err = json.Unmarshal(buf, &data)
+		assert.NoError(t, err)
```

Comment:
Should this be `require.NoError`? If deserialization fails, the subsequent `assert` will add to the noise.